src/Compilers/CSharp/Test/Semantic/Semantics/ReadOnlyStructsTests.cs

```diff
@@ -351,88 +351,128 @@ public struct S
+        public void ReadOnlyMethod_Partial_01()
+        {
+            var csharp = @"
+public partial struct S
+{
+    public int i;
+    readonly partial void M();
+}
+
+public partial struct S
+{
+    readonly partial void M()
+    {
+        i++;
     }
 }
+";
+            var comp = CreateCompilation(csharp);
+            comp.VerifyDiagnostics(
+                // (12,9): error CS1604: Cannot assign to 'i' because it is read-only
+                //         i++;
+                Diagnostic(ErrorCode.ERR_AssgReadonlyLocal, "i").WithArguments("i").WithLocation(12, 9));
+        }
 
-public struct S2
+        [Fact]
+        public void ReadOnlyMethod_Partial_02()
+        {
+            var csharp = @"
+public partial struct S
 {
-    public readonly override string ToString() => throw null;
-    public readonly override int GetHashCode() => throw null;
-    public readonly override bool Equals(object o) => throw null;
+    public int i;
+    partial void M();
+}
 
-    readonly void M()
+public partial struct S
+{
+    readonly partial void M()
     {
-        // no warnings
-        ToString();
-        GetHashCode();
-        Equals(null);
-
-        base.ToString();
-        base.GetHashCode();
-        base.Equals(null);
+        i++;
     }
 }
 ";
             var comp = CreateCompilation(csharp);
             comp.VerifyDiagnostics(
-
-                // (12,9): warning CS8655: Call to non-readonly member 'ToString' from a 'readonly' member results in an implicit copy of 'this'.
-                //         ToString();
-                Diagnostic(ErrorCode.WRN_ImplicitCopyInReadOnlyMember, "ToString").WithArguments("ToString", "this").WithLocation(12, 9),
-                // (13,9): warning CS8655: Call to non-readonly member 'GetHashCode' from a 'readonly' member results in an implicit copy of 'this'.
-                //         GetHashCode();
-                Diagnostic(ErrorCode.WRN_ImplicitCopyInReadOnlyMember, "GetHashCode").WithArguments("GetHashCode", "this").WithLocation(13, 9),
-                // (14,9): warning CS8655: Call to non-readonly member 'Equals' from a 'readonly' member results in an implicit copy of 'this'.
-                //         Equals(null);
-                Diagnostic(ErrorCode.WRN_ImplicitCopyInReadOnlyMember, "Equals").WithArguments("Equals", "this").WithLocation(14, 9));
+                // (10,27): error CS8662: Both partial method declarations must be readonly or neither may be readonly
+                //     readonly partial void M()
+                Diagnostic(ErrorCode.ERR_PartialMethodReadOnlyDifference, "M").WithLocation(10, 27),
+                // (12,9): error CS1604: Cannot assign to 'i' because it is read-only
+                //         i++;
+                Diagnostic(ErrorCode.ERR_AssgReadonlyLocal, "i").WithArguments("i").WithLocation(12, 9));
         }
 
         [Fact]
-        public void ReadOnlyMethod_Override_AssignThis()
+        public void ReadOnlyMethod_Partial_03()
```

Comment:
>ReadOnlyMethod_Partial_03 [](start = 20, length = 25)

In this test, it looks like the partial method isn't considered readonly. I expected the declaration would win (ie. method would be considered readonly).
Consider verifying the APIs for readonly on the method symbol.

Reply:
Sounds good.

---
In reply to: [270200175](https://github.com/dotnet/roslyn/pull/34468#discussion_r270200175) [](ancestors = 270200175)

Reply:
I'm seeing some pretty strange behavior from the symbol API when there's inconsistent `readonly` modifiers on the two signatures. Not sure if I should address it or just let the compile error be sufficient.

Reply:
We worked out offline that the compiler selects the "declaration" symbol for the partial method by design when looking it up in `GetMembers()`. Therefore the symbol API behaviors we are seeing are expected.